src/learning/gmm.rs

```diff
@@ -257,17 +292,16 @@ impl GaussianMixtureModel {
 
         if let Some(ref covars) = self.model_covars {
             for cov in covars {
-                // TODO: combine these. We compute det to get the inverse.
+                // TODO: Fix this, this breaks stuff.
```

Comment:
I'm not sure there is an alternative to computing the determinant and inverse here? At the least even a more numerically stable solution would retain some of the issues we see here.